frontend/src/pages/RoadDetails.tsx

```diff
@@ -0,0 +1,21 @@
+import { useState } from 'react';
+import '../css/roadDetails.css'; // Import the CSS file
+
+import MiddleArea from '../Components/roadDetails/MiddleArea';
+import MiddleArea2 from '../Components/roadDetails/MiddleArea2';
+import TopBar from '../Components/roadDetails/TopBar';
+import PlotArea from '../Components/roadDetails/PlotArea';
+
+const RoadDetails = () => {
+  const [showMiddleArea2, setShowMiddleArea2] = useState(false);
+
+  return (
+    <div>
+      <TopBar setShowMiddleArea2={setShowMiddleArea2} />
+      {showMiddleArea2 ? <MiddleArea2 /> : <MiddleArea />}
```

Comment:
Please use the MapWrapper component for the map one and rename the other component to something more specific (maybe "RoadSurface" or something like that)